pyspark_datasources/opensky.py

```
@@ -0,0 +1,350 @@
+        
+        try:
+            response = requests.post(token_url, data=data, timeout=10)
+            response.raise_for_status()
+            token_data = response.json()
+            
+            self.access_token = token_data["access_token"]
+            expires_in = token_data.get("expires_in", 1800)
+            self.token_expires_at = current_time + expires_in - 300
+            
+        except requests.exceptions.RequestException as e:
+            raise OpenSkyAPIError(f"Failed to get access token: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)